### PR TITLE
Add Overlays

### PR DESCRIPTION
**Pull request by sircommitsalot**
Addresses https://github.com/sircommitsalot/everyozcounts/issues/1

Screen Shot 2022-10-05 at 9 09 10 PM

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,15 +1,35 @@
-const data = [{
-  'name': "Sully's Snowgoose",
-  'lat': 47.673720,
-  'lng': -122.354740,
-}];
+const data = [ {
+  name: 'Sully\'s Snowgoose',
```
Comment (gmkohler) [inline index.js:2]: do we have to live this way

File: index.js [hunk 2]
```diff
@@ -1,15 +1,35 @@
...
+  lng: -121.331291,
+  hasRealDrunks: true,
+  has16ozPours: true,
+  includesTaxInPrice: true,
+} ];
+
+const displayFacts = {
+  hasRealDrunks: 'Has Real Drunks',
+  has16ozPours: 'Has 16oz Pours',
+  includesTaxInPrice: 'Includes Tax In Price',
```
Comment (gmkohler) [inline index.js:20]: We need to workshop the data format here, but this works as a proof of concept

File: index.js [hunk 3]
```diff
@@ -23,23 +43,43 @@ const getMapInitConfig = async () => {
...
-  // The marker, positioned at Uluru
+    
+    const infoWindow = new google.maps.InfoWindow({
+      content: (
+        `<h2>${name}</h2>` +
+        '<ul>' +
+        Object.entries(facts).map(renderFact).join('') +
+        '<ul>'
+      ),
+    });
```
Comment (gmkohler) [inline index.js:77]: 🤯 this feels very low rent / prone to injection for a google API

**Review by gmkohler — APPROVED** [review]

File: index.js [hunk 4]
```diff
@@ -1,15 +1,35 @@
...
-  'lng': -122.354740,
-}];
+const data = [ {
+  name: 'Sully\'s Snowgoose',
+  lat: 47.673720,
+  lng: -122.354740,
+  hasRealDrunks: true,
+  has16ozPours: true,
+  includesTaxInPrice: true,
+}, {
```
Comment (gmkohler) [inline index.js:8]:
```suggestion
},
{
```

File: index.js [hunk 5]
```diff
@@ -23,23 +43,43 @@ const getMapInitConfig = async () => {
...
     });
-  })
-  // The marker, positioned at Uluru
+    
+    const infoWindow = new google.maps.InfoWindow({
+      content: (
+        `<h2>${name}</h2>` +
+        '<ul>' +
+        Object.entries(facts).map(renderFact).join('') +
+        '<ul>'
```
Comment (gmkohler) [inline index.js:75]: honest question, can the whole thing be a template string like line 72 is?

File: index.js [hunk 6]
```diff
@@ -23,23 +43,43 @@ const getMapInitConfig = async () => {
...
-  data.map(({ lat, lng }) => {
-    new google.maps.Marker({
-      position: { lat, lng },
+  const map = new google.maps.Map(document.getElementById('map'), mapInitConfig);
+  
+  data.map(({
+    name,
+    lat,
+    lng,
+    ...facts
```
Comment (gmkohler) [inline index.js:60]: I'm surprised this works, feels to me like a weird mix of named/ordered

Comment (jkprow-dignifi) [reply index.js:60]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#rest_property

File: index.js (hunk 5 above)
Comment (jkprow-dignifi) [reply index.js:75]: Yeah it can. I was trying to preserve indentation but my editor blew it away when I formatted.

**Comment by gmkohler** [comment]
Oh this is fine for now then.  Eventually do you plan to move to something
a little more programmatic for HTML generation?

**Comment by sircommitsalot** [comment]
@gmkohler RE: HTML generation, good question. Passing raw HTML to the Maps SDK definitely #feelsbadman but the elites at Google are probably smarter than us. If we end up having to do that a lot, yeah we should look into something that adds a few layers of indirection between us and `String.concat`

**Pull request merged**